internal/runtime/hcsv2/process.go

```diff
@@ -114,15 +114,34 @@ func (p *Process) Kill(ctx context.Context, signal syscall.Signal) (err error) {
 		trace.Int64Attribute("pid", int64(p.pid)),
 		trace.Int64Attribute("signal", int64(signal)))
 
-	if err := syscall.Kill(int(p.pid), signal); err != nil {
-		if err == syscall.ESRCH {
-			return gcserr.NewHresultError(gcserr.HrErrNotFound)
-		}
-		return err
-	}
+	// When a container contains more than one process we can fail to unblock
+	// the wait if we only signal the init process. Instead we issue a `runc
+	// kill --all` which then signals all processes in the container.
 	if p.init {
-		p.c.setExitType(signal)
+		if err := p.c.Kill(ctx, signal); err != nil {
+			// Since we use the container logic to kill here it returns a
+			// `gcserr.HrVmcomputeSystemNotFound` but we need a
+			// `gcserr.HrErrNotFound` since we are operating on a process
+			// handle.
+			_, e2 := gcserr.GetHresult(err)
+			if e2 == nil {
+				return gcserr.NewHresultError(gcserr.HrErrNotFound)
```

Comment:
Why don't we return the hr from `GetHresult` here?

Reply:
Did you read the comment above? I thought that was clear lol must not have been.

Reply:
I understand why we need to convert the error into something from `NewHresultError`. I don't understand why we need to convert every HResult into `HrErrNotFound`. We want to return not found even if the kill fails for some other reason?

Reply:
The call to Kill ONLY returns `HrVmcomputeSystemNotFound` or it doesn't return a `HRESULT` error. But I can make this explicit by adding a check for that I guess.